lambdas/services/bulk_upload_metadata_preprocessor_service.py

```diff
@@ -0,0 +1,404 @@
+            try:
+                self.s3_service.client.copy_object(
+                    Bucket=self.staging_store_bucket,
+                    CopySource={
+                        "Bucket": self.staging_store_bucket,
+                        "Key": original_file_key,
+                    },
+                    Key=new_file_key,
+                    MetadataDirective="COPY",
+                    TaggingDirective="COPY",
+                )
+            except ClientError as e:
+                error_code = e.response["Error"]["Code"]
+                if error_code == "NoSuchKey":
+                    logger.info(f"File {original_file_key} doesn't exist")
+                    return None
+                else:
+                    logger.error(
+                        f"Failed update filename for `{original_file_key}`: {e}"
+                    )
+                    return None
+
+            self.s3_service.client.delete_object(
+                Bucket=self.staging_store_bucket, Key=original_file_key
+            )
+
+        return updated_row
+
+    def move_original_metadata_file(self, file_key: str):
+        destination_key = f"{self.practice_directory}/{self.processed_folder_name}/{self.processed_date}/{METADATA_FILENAME}"
+        logger.info(
+            f"Moving original metadata file from {file_key} to {destination_key}"
+        )
+        try:
+            self.s3_service.copy_across_bucket(
+                self.staging_store_bucket,
+                file_key,
+                self.staging_store_bucket,
+                destination_key,
+            )
+            return True
+        except ClientError as e:
+            logger.error(
+                f"Failed to move metadata file '{file_key}' to '{destination_key}': {e}"
+            )
+            return False
+
+    @staticmethod
+    def extract_person_name_from_bulk_upload_file_name(
+        file_path: str,
+    ) -> tuple[str, str]:
+        document_number_expression = r".*?([\p{L}][^\d]*[\p{L}])(.*)"
+        expression_result = regex.search(
+            rf"{document_number_expression}", file_path, regex.IGNORECASE
+        )
+
+        if expression_result is None:
+            logger.info("Failed to find the person name in the file name")
+            raise InvalidFileNameException("Invalid patient name")
+
+        name = expression_result.group(1)
+        current_file_path = expression_result.group(2)
+
+        return name, current_file_path
+
+    @staticmethod
+    def extract_lloyd_george_record_from_bulk_upload_file_name(
+        file_path: str,
+    ) -> tuple[str, str]:
+        _expression = r".*?ll[oO0οՕ〇]yd.*?ge[oO0οՕ〇]rge.*?rec[oO0οՕ〇]rd(.*)"
+        lloyd_george_record = regex.search(
+            rf"{_expression}", file_path, regex.IGNORECASE
+        )
+        if lloyd_george_record is None:
+            logger.info("Failed to find Lloyd George Record")
+            raise InvalidFileNameException("Invalid Lloyd_George_Record separator")
```

Comment:
This sounds like it has failed trying to find an actual record. Could we be more explicit, for example:
```suggestion
            logger.error("Failed to extract Lloyd George Record from file name")
            raise InvalidFileNameException("Invalid Lloyd_George_Record separator")
```